Convenience method "sdk flush all"

delight opened the pull request:
- flush all -> flush broadcast, flush archives and flush temp

To raise a new Pull Request, the following prerequisites need to be met. Please tick before proceeding:

- [ ] a conversation was held in the appropriate [SDKMAN Slack](https://slack.sdkman.io) channel.
- [ ] a Github Issue was opened for this feature / bug.
- [ ] test coverage was added (Cucumber or Spock as appropriate).

This addresses Issue #XXX

marc0der commented:
> To raise a new Pull Request, the following prerequisites need to be met.

Sorry, but won't merge until this requirement is met. For one, I don't see the point of flushing the broadcast as this should only be used in extreme situations. Happy to discuss it on slack.

marc0der commented on src/test/cucumber/flush.feature:
```diff
@@ -37,3 +37,12 @@ Feature: Flush
+    Then no archives are cached
+    And no "res-1.2.0.zip" file is present in temporary storage
+    And I see "1 archive(s) flushed"
+    And I see "1 archive(s) flushed"
```
Should this duplicate line be here? And what does flushing temporary storage print to console?

marc0der commented on src/test/cucumber/flush.feature:
```diff
@@ -37,3 +37,12 @@ Feature: Flush
+  Scenario: Clear out the cached archives and the temporary space
+    Given the archive "grails-1.3.9.zip" has been cached
+    And the file "res-1.2.0.zip" in temporary storage
+    When I enter "sdk flush all"
```
I'm wondering if `all` isn't redundant. Following the way the rest of the CLI works, we try making things as lean as possible. How about a simple `sdk flush` being responsible for flushing _all_. wdyt?

delight replied:
Hi Marc, I personally like your idea of a `sdk flush` being used to flush temp and archives.
There are just 2 questions coming into my mind:
1) Is it what a user would expect to happen when by simply typing `sdk flush ` ? If so, yes, let us do it that way.
2) What should the help page display ? 
`       flush             [archives|temp|broadcast|version]`
instead of 
`        flush             <broadcast|archives|temp>`

About your question:

> Should this duplicate line be here? And what does flushing temporary storage print to console?

The line isn't duplicate if both temp and archives having 'data' the new all `sdk flush all` would print 2 lines: 
```
1 archive(s) flushed, freeing 13M       /home/koko/.sdkman/archives.
1 archive(s) flushed, freeing 16K       /home/koko/.sdkman/tmp.
```
So the I tried picking this 2 lines with cucumber - might have been the wrong atempt.

marc0der commented on src/main/bash/sdkman-help.sh:
```diff
@@ -35,7 +35,7 @@ function __sdk_help {
 	__sdkman_echo_no_colour "       selfupdate        [force]"
 	__sdkman_echo_no_colour "       update"
-	__sdkman_echo_no_colour "       flush             <broadcast|archives|temp>"
+	__sdkman_echo_no_colour "       flush             <broadcast|archives|temp|all>"
```
If `all` is redundant, we could make this:

    flush           [broadcast|archives|temp]

BTW, I'll remove the flushing of broadcast in a subsequent PR as it is actually no longer required.

delight replied:
Oh sorry I should have read further :see_no_evil:

marc0der commented on src/test/cucumber/flush.feature:
```diff
@@ -37,3 +37,12 @@ Feature: Flush
     Then no "res-1.2.0.zip" file is present in temporary storage
     And I see "1 archive(s) flushed"
+
+  Scenario: Clear out the cached archives and the temporary space
```
temporary storage?

delight replied:
ok I'll implement those advice and push them :+1:

delight replied:
by the way I was just orienting myself from the tests already written :-)

>  Scenario: Clear out the temporary space

vpavic commented:
Any updates on this?

After yet another `sdk flush archives && sdk flush temp`, I came here looking to see whether `flush all` has been proposed and stumbled upon this PR.

delight commented:
Hi @vpavic 

thb I don't know about the status. I implemented everything as requested, changed it accordingly to the discussion we had with @marc0der but the pull request never got merged. I can rebase it on the current development if wanted.

Cheers,
   Kosta

marc0der commented:
@delight apologies, this must have slipped under the radar. I'll try resolve the conflicts locally and merge it today.

marc0der commented:
Merge conflicts are now resolved and I'll release later today.

vpavic commented:
Thanks for the quick turnaround @marc0der! :+1:

marc0der commented:
No worries, it was my oversight. This has now been released:
https://github.com/sdkman/sdkman-cli/releases/tag/5.9.1
